docs/authoring.md

```diff
@@ -259,3 +224,112 @@ There are some template parameters that have a special meaning:
 |Name|Description|
 |---|---|
 |`server`| It is used to let the template know which server from the AsyncAPI specification file you want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. Use this parameter in case your template relies on users' information about what server from the specification file they want to use during generation. You also need this parameter if you want to use `server.protocol` notation within `conditionalFiles` configuration option. Once you decide to specify this parameter for your template, it is recommended you make it a mandatory parameter otherwise feature like `conditionalFiles` is not going to work if your users do not use this parameter obligatory.
+
+
+## React
+[React](https://reactjs.org) is the render engine which we strongly suggest you should use for any new templates. The only reason it is not the default render engine is to stay backward compatibility.
+* It enables the possiblity of [debugging](#debugging-react-template) your template (this is not possible with Nunjucks).
+* It provides better error stack traces once something is not working.
+* Provides better support for separating code into more manageable chunks/components.
+* The readability of the template are much better then Nunjucks syntax.
+* Better tool support for development.
+
+When writing react templates you decide whether to use CommonJS, ES5 or ES6 modules since everything is bundles together before the rendering process takes over. We use our own React renderer which can be found in the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk). 
+There you can find information about how the renderer works or how we transpile your template files.
+
+Your React template always require 3 dependencies `react`, `source-map-support` and `@asyncapi/generator-react-sdk`. The `react` dependency is required to declare React components inside the template. `source-map-support` ensure that the transpiled files are mapped to the actual template files, this is only to enable correct error stack trace. `@asyncapi/generator-react-sdk` is required to access the `File` component which is required as a root component for a file to be rendered. Furthermore it provides some common components to make your development easier, like `Text` or `Indent`.
+
+Lets consider a basic react template as the one below called `MyTemplate.js`:
+```js
+import React from "react";
+import { File, Text } from "@asyncapi/generator-react-sdk";
+export default function({ asyncapi, params, originalAsyncAPI}) {
```

Comment:
> export default function({ asyncapi, params, originalAsyncAPI}) {

I wonder if exporting anonymous functions is the recommended pattern with React. Also, wouldn't it make things harder to debug?

Reply:
Good question, @magicmatatjahu I think we need your input here 😄

Reply:
At one time, default exports were recommended by the community later that changed 😄 

Return `default` function is crucial for DX and in which way works at this moment the Generator. Why? Because by default export you don't have to name function by `SchemaTemplate` or another predefined name for Generator. We just import the file and then perform some actions to render the template to file. If we enable export by names, then we have a issue: which function we should process? Let's check the example:

```js
export function SomeTemplate({}) {
  // ...implementation
}

export function SomeTemplate2({}) {
  // ...implementation
}
```

which we should choose? As I remember, we didn't discuss on the possibility of rendering several different templates in one file, the difference may be the rendering of an array of templates, which differs only on input data (yes, I am a hypocrite hah 😄).

Reply:
I don't mean removing `default` but instead adding a function name:

```suggestion
export default function Example({ asyncapi, params, originalAsyncAPI}) {
```

Reply:
Ahhh, ok. Mea culpa. It's should recommended, so we should change the example, but it's not mandatory.

Reply:
@magicmatatjahu can we mark this as resolved?